xds/src/test/java/io/grpc/xds/ClientXdsClientDataTest.java

```diff
@@ -647,13 +656,153 @@ public void parseHttpFilter_unsupportedAndRequired() {
+                                        .addPrincipals(Principal.newBuilder().setAny(true))
+                                        .addPermissions(Permission.newBuilder().setAny(true))
+                                        .build())
+                                .build())
+                        .build()))
+            .build();
+    FilterConfig config = ClientXdsClient.parseHttpFilter(
+        httpFilter, filterRegistry, false /* isForClient */).getStruct();
+    assertThat(config).isInstanceOf(RbacConfig.class);
+  }
+
+  @Test
+  public void parseHttpFilter_rbacConfigUnsupportedForClient() {
+    filterRegistry.register(RbacFilter.INSTANCE);
+    HttpFilter httpFilter =
+        HttpFilter.newBuilder()
+            .setIsOptional(false)
+            .setName("envoy.auth")
+            .setTypedConfig(
+                Any.pack(
+                    io.envoyproxy.envoy.extensions.filters.http.rbac.v3.RBAC.newBuilder()
+                        .setRules(
+                            RBAC.newBuilder()
+                                .setAction(Action.ALLOW)
+                                .putPolicies(
+                                    "allow-all",
+                                    Policy.newBuilder()
+                                        .addPrincipals(Principal.newBuilder().setAny(true))
+                                        .addPermissions(Permission.newBuilder().setAny(true))
+                                        .build())
+                                .build())
+                        .build()))
+            .build();
+    StructOrError<FilterConfig> config =
+        ClientXdsClient.parseHttpFilter(httpFilter, filterRegistry, true /* isForClient */);
+    assertThat(config.getErrorDetail()).isEqualTo(
+        "HttpFilter [envoy.auth](" + RbacFilter.TYPE_URL + ") is required but "
+            + "unsupported for client");
+  }
+
+  @Test
+  public void parseOverrideRbacFilterConfig() {
+    filterRegistry.register(RbacFilter.INSTANCE);
+    RBACPerRoute rbacPerRoute =
+        RBACPerRoute.newBuilder()
+            .setRbac(
+                io.envoyproxy.envoy.extensions.filters.http.rbac.v3.RBAC.newBuilder()
+                    .setRules(
+                        RBAC.newBuilder()
+                            .setAction(Action.ALLOW)
+                            .putPolicies(
+                                "allow-all",
+                                Policy.newBuilder()
+                                    .addPrincipals(Principal.newBuilder().setAny(true))
+                                    .addPermissions(Permission.newBuilder().setAny(true))
+                            .build())))
+            .build();
+    Map<String, Any> configOverrides = ImmutableMap.of("envoy.auth", Any.pack(rbacPerRoute));
+    Map<String, FilterConfig> parsedConfigs =
+        ClientXdsClient.parseOverrideFilterConfigs(configOverrides, filterRegistry).getStruct();
+    assertThat(parsedConfigs).hasSize(1);
```

Comment:
This test might be flaky as it fails for me (when I built from this branch):
```
io.grpc.xds.ClientXdsClientDataTest > parseOverrideRbacFilterConfig FAILED
    java.lang.NullPointerException
        at com.google.common.truth.MapSubject.hasSize(MapSubject.java:98)
        at io.grpc.xds.ClientXdsClientDataTest.parseOverrideRbacFilterConfig(ClientXdsClientDataTest.java:798)

```

Tried twice and both times it failed.

Reply:
Did you pull master? You need to include #8262.

Reply:
Alright, I just synced this PR with master. You can try it again by pulling my branch.

Reply:
I completely deleted the branch and pulled in your branch from your fork. That should have been sufficient?

Reply:
Okay it passes now after resyncing